src/store/api/contentful.ts

```diff
@@ -0,0 +1,81 @@
+import { createClient, type EntryFieldTypes, type Entry } from 'contentful';
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+type FromSkeleton<T extends ContentfulSkeletons> = Entry<T, 'WITHOUT_UNRESOLVABLE_LINKS'>['fields'];
+
+type AdSkeleton = {
+  contentTypeId: 'ads',
+  fields: {
+    title: EntryFieldTypes.Text,
+    description: EntryFieldTypes.Text,
+    fullscreen: EntryFieldTypes.Boolean,
+    poster: EntryFieldTypes.AssetLink
+  }
+};
+type Ad = FromSkeleton<AdSkeleton>;
+
+type BoardMessageSkeleton = {
+  contentTypeId: 'board-message',
+  fields: {
+    message: EntryFieldTypes.Text
+  }
+};
+type BoardMessage = FromSkeleton<BoardMessageSkeleton>;
+
+type QuoteSkeleton = {
+  contentTypeId: 'quotes',
```

Comment:
normally in types you specify the type, not the value. 

I think you should create something like this:

```ts
interface ContenfulRequest<T> {
contentTypeId: string;
fields: T;
}

type BoardMessage =  {
message: EntryFieldTypes.Text
}
```

Now you don't need the magic from skeleton function, can type request with `ContenfulRequest<BoardMessage>` and most importantly don't repeat yourself